pkg/proto/nbcontract/v1/authconfig.proto

```diff
@@ -0,0 +1,13 @@
+syntax = "proto3";
+package nbcontract.v1;
+
+// Auth Config fields stored in azure.json used by cloud-provider-azure
+message AuthConfig {
+  string target_cloud = 1; // set to cloud, default to AzurePublicCloud
+  string tenant_id = 2;
+  string subscription_id = 3;
+  string service_principal_id = 4; // set to aadClientId
+  string service_principal_secret = 5; // set to aadClientSecret
```

Comment:
At least these (and possibly others) can be unspecified. What pattern did we settle on for handling optional strings?

Reply:
I think we decided that unless we need to change the default value of just an empty string or need to handle the nil case specially, then there is no need to specify optional. In this case, not specifying it should have an empty string value so i excluded the optional.